test-d/foundry/common/data/data.mjs/measuredTemplateData.test-d.ts

```diff
@@ -1,6 +1,9 @@
-import { expectType } from 'tsd';
+import { expectError, expectType } from 'tsd';
 
-expectType<foundry.data.MeasuredTemplateData>(new foundry.data.MeasuredTemplateData());
+expectError<foundry.data.MeasuredTemplateData>(new foundry.data.MeasuredTemplateData());
+expectError<foundry.data.MeasuredTemplateData>(new foundry.data.MeasuredTemplateData({ x: 100, y: 100 }));
```

Comment:
Please have a look at the two failing tests above. I removed both again, because I am not deep enough in the source to know whether this calls can be prevented, but both will fail when called in foundry:
```
data.mjs:298 Uncaught TypeError: Cannot read property 'parent' of null
    at MeasuredTemplateData._validateDocument (data.mjs:980)
    at MeasuredTemplateData.validate (data.mjs:295)
    at new DocumentData (data.mjs:48)
    at new MeasuredTemplateData (data.mjs:946)
    at <anonymous>:1:1
```

Reply:
Currently, the constructor parameters are optional in DocumentData. I suppose, we would need to make them mandatory for these tests to pass. But I think there is document data, where it’s valid for them to be optional, so I don’t think it’s the Right thing in the general case… so we have to options:
1. accept the current behavior (users don’t get any help from the types to prevent these runtime errors)
2. Override the constructor for MeasuredTenplateData in the types (even if it’s not done in foundry) to make the parameters mandatory

Both options are fine IMO. If we go for the second one, we should add a remark to the constructor why it hast been added.